common.py

```diff
@@ -0,0 +1,10 @@
+from twisted.web.http_headers import Headers
+
+
+def listify_headers(headers):
+    """Twisted agent requires header values as lists
+    """
+    for key, val in headers.iteritems():
+        if not isinstance(val, list):
+            headers[key] = [val]
+    return Headers(headers)
```

Comment:
This is such a bizarre interface for `Headers`. I have no idea why Twisted would do this.